File: 524. Longest Word in Dictionary through Deleting/Python/Longest Word in Dictionary through Deleting.py
class Solution(object):
    def findLongestWord(self, s, d):
        """
        :type s: str
        :type d: List[str]
        :rtype: str
        """
        d.sort(cmp=lambda a, b: cmp(a, b) if len(a) == len(b) else len(b) - len(a))
        for word in d:
            i = 0
            for letter in s:
                if letter == word[i]:
                    i += 1
                if i == len(word):
                    break
            if i == len(word):
                return word
        return ''

File: 524. Longest Word in Dictionary through Deleting/Java/Longest Word in Dictionary through Deleting.java
public class Solution {
    public String findLongestWord(String s, List<String> d) {
        Collections.sort(d, new Comparator<String>() {
            @Override
            public int compare(String o1, String o2) {
                if (o1.length() == o2.length()) {
                    return o1.compareTo(o2);
                } else {
                    return o2.length() - o1.length();
                }
            }
        });
        for (String word : d) {
            int i = 0;
            for (char letter : s.toCharArray()) {
                if (letter == word.charAt(i)) {
                    i++;
                }
                if (i == word.length()) {
                    break;
                }
            }
            if (i == word.length()) {
                return word;
            }
        }
        return "";
    }
}